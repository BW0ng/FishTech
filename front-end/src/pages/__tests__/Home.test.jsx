import { mount } from 'enzyme';
import HomePage from '../Home';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Home Page', () => {
    it ('renders correctly', () => {
        const component = renderer.create(<HomePage />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // TODO Need to figure out how to get history from props. Props is currently {}
    // it('Submit button redirects to lookup page', () => {
    //     const mockPush = jest.fn();
    //     jest.mock('react-router-dom', () => ({
    //     useHistory: () => {
    //         const push = () => mockPush ();
    //         return { push };
    //     },
    //     }));

    //     const wrapper = mount(<HomePage/>);

    //     wrapper.find('#whois-input').text = "Input Text";

    //     // Find the button and call the onClick handler
    //     wrapper.find('button').simulate('click');
        
    //     expect(mockPush).toBeCalledWith(['/lookup', {domainName: "Input Text"}]);
    // })
})