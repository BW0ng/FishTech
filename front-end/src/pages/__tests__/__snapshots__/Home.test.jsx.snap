// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders correctly 1`] = `
<div
  class="home-page"
>
  <h1>
    WHOIS Lookup
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      class="form-group"
    >
      <input
        aria-describedby="emailHelp"
        class="form-control"
        id="whois-input"
        placeholder="Enter a domain name or IP address to lookup"
        type="whois"
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
